### ISSUE TITLE
[2][2.0.190410-rc1] Maintain Language

### ISSUE THREAD

**Issue by sschiessl-bcp** [bitshares/bitshares-ui#2558]
Translations are done through transifex, please find the file here

https://www.transifex.com/bitshares/bitshares-ui-1

Bounties:
 - Translate per Word: Complete Chinese, Russian, German and Spanish. When assigning a language, please state before how many words are missing. Bounty $0.15/word, according to https://search.proz.com/employers/rates plus BitShares knowledge as bonus, open to discussion if that is unsuitable

 - max. 4 each: Review Chinese, Russian, German and Spanish. Must be different person than the one doing the translation

Please voice interest in other languages that are already partly translated and we discuss bounties. Every effort through this issue must be approved beforehand.

**Comment by sschiessl-bcp** [comment]
@ruslansalikhov

**Comment by sschiessl-bcp** [comment]
@TechsUsInc

**Comment by bieglich** [comment]
I'd be interested in reviewing german.

**Comment by TechsUsInc** [comment]
@sschiessl-bcp 
Request to complete Japanese

**Comment by shulthz** [comment]
@sschiessl-bcp
Chinese translation finished.

**Comment by froooze** [comment]
@sschiessl-bcp: I found some errors in German translation, can edit and afterwards @bieglich makes the review?

**Comment by TechsUsInc** [comment]
Spanish translated;reviewed

**Comment by ruslansalikhov** [comment]
@sschiessl-bcp blckchnd will continue Russian

**Comment by TechsUsInc** [comment]
Chinese reviewed for 2nd time. Transifex doesn't seem to reflect 30% Spanish/Russian/Chinese translations previously translated/reviewed

**Comment by froooze** [comment]
> Chinese reviewed for 2nd time. Transifex doesn't seem to reflect 30% Spanish/Russian/Chinese translations previously translated/reviewed

The percentage numbers can not be trusted on transifex.

**Comment by sschiessl-bcp** [comment]
> @sschiessl-bcp: I found some errors in German translation, can I translate German and afterwards @bieglich makes the review?

Sounds perfect, go ahead. Please post the word count.

**Comment by sschiessl-bcp** [comment]
> @sschiessl-bcp
> Request to complete Japanese
> Request to complete Russian

I might come back to you on Japanese, Russian will be done by BLCKCHND (since they did that always in the past)

**Comment by sschiessl-bcp** [comment]
> Spanish translated;reviewed

Please state the word count. Did you approve the translations in Transifex (there is an explicit review step)?

**Comment by sschiessl-bcp** [comment]
> @sschiessl-bcp
> Chinese translation finished.

Please state your word count.

**Comment by froooze** [comment]
@sschiessl-bcp: What happens to the unfilled brackets? 
Some look important others not so much.

**Comment by sschiessl-bcp** [comment]
Can you show an example plz?

**Comment by froooze** [comment]
@sschiessl-bcp :

```
"orderbook_autoscroll": "",
        "orderbook_grouping": "",
        "orderbook_reverse": "",
```
```
   "right": "",
     "scrollbars_hide": "",
```
` "scaled": "",`
```
"scaled_orders": {
        "action": {
            "buy": "",
            "sell": "",
            "title": ""
        },
        "distribution": {
            "flat": "",
            "title": ""
        },
        "fee": "",
        "market_fee": "",
        "order_count": "",
        "order_s": "",
        "please_log_in": "",
        "preview_table": {
            "price": ""
        },
        "price_lower": "",
        "price_upper": "",
        "quantity": "",
        "title": "",
        "total": ""
```
Is there a easy way to find out about words by comparing two files?

**Comment by sschiessl-bcp** [comment]
Transifex states the remaining words to be translated here
![image](https://user-images.githubusercontent.com/33128181/55537564-cb876800-56bc-11e9-909d-076445efb4de.png)

But how can that be for german? I thought you translated everything already? I didn't check in Transifex back then, but it appears there is lots missing in there.

**Comment by froooze** [comment]
I translated every phrase, which was in English to German, but there are a lot of phrases, which have no English entry and are empty, like I posted above. 

The number of untranslated words doesn't make any sense. I see this as a bug in the UI.

**Comment by sschiessl-bcp** [comment]
Here is an example:
https://github.com/bitshares/bitshares-ui/blob/develop/app/assets/locales/locale-en.json#L209
https://github.com/bitshares/bitshares-ui/blob/develop/app/assets/locales/locale-de.json#L209

https://github.com/bitshares/bitshares-ui/blob/develop/app/assets/locales/locale-en.json#L389
https://github.com/bitshares/bitshares-ui/blob/develop/app/assets/locales/locale-de.json#L389

**Comment by sschiessl-bcp** [comment]
I think I found the missing step ...

Somehow your translation got only added as suggestion in transifex
![image](https://user-images.githubusercontent.com/33128181/55609204-6d23bd80-5780-11e9-9f14-bee8ff53b545.png)


But for others that is also missing
![image](https://user-images.githubusercontent.com/33128181/55609281-93495d80-5780-11e9-8cec-e60df14b0259.png)

I would suggest that bieglich does a review sweep now and approves all suggestions from you that are in there, afterwards you (@froooze) can translate all remaining words. That is likely the reason why you see % that do not match what you have already done.

**Comment by froooze** [comment]
When I select German on transifex I get this dialog.
![transifex](https://user-images.githubusercontent.com/40874087/55635403-dd523380-57c0-11e9-8816-a96b9aaa7a80.png)
I went with *Download file to translate* and got this:
[for_translation_bitshares-ui-1_locale-enjson_de(2).json.zip](https://github.com/bitshares/bitshares-ui/files/3048405/for_translation_bitshares-ui-1_locale-enjson_de.2.json.zip)

The file is missing some strings, because 209 and 389 are empty and the English text is missing.

**Comment by sschiessl-bcp** [comment]
> The file is missing some strings, because 209 and 389 are empty and the English text is missing.

English text is always missing because english duplicates are removed before upload. English shuold only be present in english language file.

**Comment by DimaBLCKCHND** [comment]
we spent 16 hours on Russian translation and review of mistakes that cause by previous translators that made a lot of mistakes and nobody told us about it 
2 weeks ago
still, have nothing paid

**Comment by DimaBLCKCHND** [comment]
Due to an error by Stephan in handling out transifex translation permissions, someone without knowledge of Russian got translation permissions and did a lot of really bad translations, that none told us about.
We've spent 16 hours retranslation and reviewing.

And after 2+ weeks since the work we are yet to be paid.

**Comment by sschiessl-bcp** [comment]
> Due to an error by Stephan in handling out transifex translation permissions, someone without knowledge of Russian got translation permissions and did a lot of really bad translations, that none told us about.
> We've spent 17 hours retranslation and reviewing.
> 
> And after 2+ weeks since the work we are yet to be paid.

See my comments below https://github.com/bitshares/bitshares-ui/issues/2586#issuecomment-490830021
